src/Raven.Server/Documents/Sharding/ShardedDocumentDatabase.cs

```diff
@@ -141,8 +141,9 @@ public void HandleReshardingChanges()
                 // cleanup values
                 t = DeleteBucketAsync(process.Bucket, process.MigrationIndex, process.LastSourceChangeVector);
 
-                t.ContinueWith(_ =>
+                t?.ContinueWith(_ =>
                 {
+                    t.RunSynchronously();
```

Comment:
What was the issue here and how does it fix it?

TBH I'm not familiar with this one, 
but it doesn't looks correct and actually I expect it to throw because `t` is already completed here, no?